x/slashing/abci.go

```diff
@@ -23,3 +24,12 @@ func BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock, k keeper.Keeper)
 		k.HandleValidatorSignature(ctx, voteInfo.Validator.Address, voteInfo.Validator.Power, voteInfo.SignedLastBlock)
 	}
 }
+
+// Called every block, update validator set
+func EndBlocker(ctx sdk.Context, k keeper.Keeper, stakingKeeper stakingkeeper.Keeper) []abci.ValidatorUpdate {
+	EpochInterval := stakingKeeper.GetParams(ctx).EpochInterval
+	if ctx.BlockHeight()%EpochInterval == 0 {
+		k.ExecuteEpoch(ctx)
+	}
+	return []abci.ValidatorUpdate{}
```

Comment:
Is this right? Seems like it should have much more to say about validator updates? If not, can a comment be added that all validator updates are applied within staking?

Reply:
Added comment like below
`	// ValidatorSet update is done on staking module endblocker
	// Therefore endblocker of slashing module should be configured to run before staking module endblocker.`

Reply:
makes sense, thanks!